numpy/core/_add_newdocs.py

```diff
@@ -1333,6 +1333,25 @@
         this rule may result in the last element of `out` being greater
         than `stop`.
 
+    Warnings
+    --------
+    The length of the output might not be numerically stable.
+
+    Another stability issue is due to the internal implementation of `numpy.arange`.
+    The actual step value used to populate the array is
+    ``dtype(start + step) - dtype(start)`` and not `step`. Precision loss
+    can occur here, due to casting or due to using floating points when
+    `start` is much larger than `step`. This can lead to unexpected
+    behaviour. For example::
+
+      >>> np.arange(0, 5, 0.5, dtype=int)
+      array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+      >>> np.arange(-3, 3, 0.5, dtype=int)
+      array([-3, -2, -1,  0,  1,  2,  3,  4,  5,  6,  7,  8])
+
+    A workaround to these problems is to use `numpy.linspace`, which has a
+    numerically stable implementation.
```

Comment:
Agreed, though I think this could be better phrased - something like "linspace is preferred" rather than calling it a "workaround".